lib/Dialect/DC/Transforms/Analysis.cpp

```diff
@@ -0,0 +1,238 @@
+  }
+  return res;
+}
+
+DotNode createDCNode(
+    Operation &op,
+    llvm::SmallVector<std::pair<mlir::Value, std::string>> &valuesMap) {
+
+  auto tokenFlag = false;
+  if (isa<dc::UnpackOp>(op))
+    tokenFlag = true;
+
+  DotNode n = {op.getName().stripDialect().str(),
+               valueToName(op.getOperands(), valuesMap, false),
+               valueToName(op.getOperands(), valuesMap, tokenFlag)};
+
+  return n;
+}
+
+DotNode createCombNode(
+    Operation &op,
+    llvm::SmallVector<std::pair<mlir::Value, std::string>> &valuesMap) {
+
+  DotNode n = {"", valueToName(op.getOperands(), valuesMap, false),
+               valueToName(op.getOperands(), valuesMap, false)};
+
+  TypeSwitch<mlir::Operation *>(&op)
+      .Case<comb::AddOp>([&](auto op) { n.nodeType = "+"; })
+      .Case<comb::AndOp>([&](auto op) { n.nodeType = "&&"; })
+      .Case<comb::XorOp>([&](auto op) { n.nodeType = "^"; })
+      .Case<comb::OrOp>([&](auto op) { n.nodeType = "||"; })
+      .Case<comb::MulOp>([&](auto op) { n.nodeType = "x"; })
+      .Case<comb::MuxOp>([&](auto op) { n.nodeType = "mux"; })
+      .Case<comb::ICmpOp>([&](auto op) {
+        switch (op.getPredicate()) {
+        case circt::comb::ICmpPredicate::eq: {
+          n.nodeType = "==";
+          break;
+        }
+        case circt::comb::ICmpPredicate::ne: {
+          n.nodeType = "!=";
+          break;
+        }
+        case circt::comb::ICmpPredicate::sgt: {
+          n.nodeType = ">";
+          break;
+        }
+        case circt::comb::ICmpPredicate::sge: {
+          n.nodeType = ">=";
+          break;
+        }
+        case circt::comb::ICmpPredicate::slt: {
+          n.nodeType = "<";
+          break;
+        }
+        case circt::comb::ICmpPredicate::sle: {
+          n.nodeType = "<=";
+          break;
+        }
+        case circt::comb::ICmpPredicate::ugt: {
+          n.nodeType = ">";
+          break;
+        }
+        case circt::comb::ICmpPredicate::uge: {
+          n.nodeType = ">=";
+          break;
+        }
+        case circt::comb::ICmpPredicate::ult: {
+          n.nodeType = "<=";
+          break;
+        }
+        case circt::comb::ICmpPredicate::ule: {
+          n.nodeType = "<";
+          break;
+        }
+        case circt::comb::ICmpPredicate::ceq: {
+          n.nodeType = "==";
+          break;
+        }
+        case circt::comb::ICmpPredicate::cne: {
+          n.nodeType = "!=";
+          break;
+        }
+        case circt::comb::ICmpPredicate::weq: {
+          n.nodeType = "==";
+          break;
+        }
+        case circt::comb::ICmpPredicate::wne: {
+          n.nodeType = "!=";
+          break;
+        }
+        }
+      });
+  return n;
+}
+
+namespace {
+/// Emit the dot nodes
+struct DCDotPrintPass : public circt::dc::impl::DCDotPrintBase<DCDotPrintPass> {
+  void runOnOperation() override {
+
+    ModuleOp op = getOperation();
+
+    std::error_code ec;
+    llvm::raw_fd_ostream outfile("dc-out.dot", ec);
```

Comment:
It might be beneficial when this file name could be provided as a pass parameter

Reply:
Thank you! Would it be better to pass it as a parameter or just print to std out (as for example [HW](https://github.com/llvm/circt/blob/main/lib/Dialect/HW/Transforms/HWPrintInstanceGraph.cpp) does)?

Reply:
Probably pass it as a parameter.